src/condor_startd.V6/ResAttributes.cpp

```diff
@@ -659,7 +659,7 @@ static int d_log_devids = D_FULLDEBUG;
 
 const char * MachAttributes::DumpDevIds(std::string & buf, const char * tag, const char * sep)
 {
-	for (auto f : m_machres_nft_map) {
+	for (auto & f : m_machres_nft_map) {
```

Comment:
How about we use destructured bindings here (and elsewhere in this PR) to make things clearer, e.g.:
```
for (auto &[name, nft] : m_machres_nft_map) {
     if (tag && strcasecmp(tag, name.c_str())) continue ...
```